zips/zip-0228.rst

```diff
@@ -2,12 +2,539 @@
+
+Spend Authorization Signature Changes
+-------------------------------------
+
+The spend authorization signature is computed in the manner specified in Section 4.15 of the protocol specification [#protocol-spendauthsig]_, except that the signature is generated over the Action Group Hash, :math:`\mathsf{AGHash}`, which is defined to be
+:math:`\mathsf{AGHash} := \mathsf{BLAKE2b}\texttt{-}\mathsf{256}(\texttt{"ZcashActionGroup"}, \mathsf{preAuthActionGroup})`.
+
+This change ensures that no changes are required in the Action statement [#zip-0226-circuit-statement]_ to be proved.
+
+SIGHASH Computation Changes
+---------------------------
+
+The addition of Action Groups to the Zcash transaction as another level of abstraction requires a change in the way transactions are hashed to compute the SIGHASH in ZIP 244, specifically in the ``orchard_digest`` section [#zip-0244-orchard-digest]_.
+We update the structure of the ``orchard_digest`` component of the SIGHASH as follows::
+
+    orchard_digest
+    ├── orchard_action_groups_digest
+    │   ├── orchard_actions_compact_digest
+    │   ├── orchard_actions_memos_digest
+    │   ├── orchard_actions_noncompact_digest
+    │   ├── flagsOrchard
+    │   ├── anchorOrchard
+    │   └── timeLimit
+    └── valueBalanceOrchard
+
+T.4: orchard_digest
+```````````````````
+In the case that Orchard actions are present in the transaction, this digest is
+a BLAKE2b-256 hash of the following values ::
+
+    T.4a: orchard_action_groups_digest   (32-byte hash output)
+    T.4b: valueBalanceOrchard            (64-bit signed little-endian)
+
+The personalization field of this hash is set to::
+
+  "ZTxIdOrchardHash"
+
+In the case that the transaction has no Orchard actions, ``orchard_digest`` is ::
+
+    BLAKE2b-256("ZTxIdOrchardHash", [])
+
+T.4a: orchard_action_groups_digest
+''''''''''''''''''''''''''''''''''
+
+A BLAKE2b-256 hash of the subset of Orchard Action Groups information for all Orchard Action Groups belonging to the transaction.
+For each Action Group, the following elements are included in the hash::
+
+    T.4a.i  : orchard_actions_compact_digest      (32-byte hash output)
+    T.4a.ii : orchard_actions_memos_digest        (32-byte hash output)
+    T.4a.iii: orchard_actions_noncompact_digest   (32-byte hash output)
+    T.4a.iv : flagsOrchard                        (1 byte)
+    T.4a.v  : anchorOrchard                       (32 bytes)
+    T.4a.vi : timeLimit                           (4 bytes)
+
+The content of T.4a.i, T.4a.ii, and T.4a.iii are the same as T.4a, T.4b, and T.4c of the ``txid_digest`` defined in ZIP 244 [#zip-0244-orchard-digest]_, over the Actions in the Action Group, and therefore are not expanded on for the sake of brevity.
+
+The personalization field of this hash is set to::
+
+  "ZTxIdOrcActGHash"
+
+In the case that the transaction has no Action Groups, ``orchard_action_groups_digest`` is ::
+
+    BLAKE2b-256("ZTxIdOrcActGHash", [])
+
+Binding Signature Changes
```

Comment:
There is a change to how the `bsk` is computed for this, since it is from a Swap Order -- this needs to be described somewhere, and the best place seemed here.